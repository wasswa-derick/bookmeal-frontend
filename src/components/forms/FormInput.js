import React from "react";
import PropTypes from "prop-types";
import InlineError from "../InlineError";

const FormInput = ({ value, type, name, label, onChange, error }) => (
  <div className="form-group">
    <label htmlFor={name}>{label}</label>
    <input
      className={error ? "form-control is-invalid" : "form-control"}
      type={type}
      id={name}
      autoComplete="off"
      value={value}
      name={name}
      onChange={onChange}
    />
    {error && <InlineError text={error} />}
  </div>
);

FormInput.propTypes = {
  value: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  error: PropTypes.string
};

FormInput.defaultProps = {
  error: undefined
};

export default FormInput;
