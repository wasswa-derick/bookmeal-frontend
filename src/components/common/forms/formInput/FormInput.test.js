import React from "react";
import { shallow } from "enzyme";
import FormInput from "./FormInput";

describe("FormInput", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <FormInput
        onChange={jest.fn}
        name="test"
        type="text"
        label="Test"
        value=""
        error=""
      />
    );
  });

  it("renders <input />", () => {
    expect(wrapper.find("input").length).toEqual(1);
  });

  it("displays the right label", () => {
    const text = wrapper.find("label").text();
    expect(text).toEqual("Test");
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
