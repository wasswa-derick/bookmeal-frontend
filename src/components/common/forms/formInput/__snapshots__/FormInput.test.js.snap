// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="test"
  >
    Test
  </label>
  <input
    autoComplete="off"
    className="form-control"
    id="test"
    name="test"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
